lib/solidus_friendly_promotions/testing_support/factories/friendly_shipping_rate_discount_factory.rb

```diff
@@ -4,7 +4,11 @@
   factory :friendly_shipping_rate_discount, class: "SolidusFriendlyPromotions::ShippingRateDiscount" do
     amount { BigDecimal("-4.00") }
     shipping_rate
-    promotion_action { SolidusFriendlyPromotions::Actions::AdjustShipment.new }
+    promotion_action do
+      promotion = create(:friendly_promotion)
+      hundred_percent = SolidusFriendlyPromotions::Calculators::Percent.new(preferred_percent: 100)
+      SolidusFriendlyPromotions::Actions::AdjustShipment.create!(promotion: promotion, calculator: hundred_percent)
+    end
     label { "10% off" }
   end
```

Comment:
Nit: the amount, label, and calculator percentage do not match which leads to confusion

Reply:
Fixed I think

Reply:
Mostly... line 9 var name is hundred_percent still